### PR TITLE
Changes to allow it to work in Brussel too

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by wpoely86** [comment]
No jenkins?

**Comment by itkovian** [comment]
Jenkins is having issues, so we need to manually do something. @stdweird knows what :)

File: bin/dquota.py [hunk 1]
```diff
@@ -140,12 +142,13 @@ def main():
                 stats["%s_users" % (storage_name,)] = 0
                 logger.debug("storage_name %s found no users who are exceeding their quota" % storage_name)
 
-    except Exception, err:
+    except Exception as err:
         logger.exception("critical exception caught: %s" % (err))
         opts.critical("Script failed in a horrible way")
         sys.exit(NAGIOS_EXIT_CRITICAL)
```
Comment (stdweird) [inline bin/dquota.py]: so this is wrong, should exit with `NAGIOS_EXIT_CRITICAL[0]` (and all pplaces where this is called)

File: lib/vsc/filesystem/quota/tools.py [hunk 2]
```diff
@@ -107,6 +109,10 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         return True
 
     def push(self, storage_name, payload):
+        if storage_name not in self.payload:
+            logging.error("Adding payload for unknown storage: %s vs %s", storage_name, self.storage_name)
```
Comment (stdweird) [inline lib/vsc/filesystem/quota/tools.py]: `Adding` -> `Can't add`?

File: lib/vsc/filesystem/quota/tools.py [hunk 3]
```diff
@@ -115,11 +121,46 @@ def push(self, storage_name, payload):
...
+            "files_doubt": quota.files_doubt,
+            "files_expired": quota.files_expired[0],
+            "files_remaining": quota.files_expired[1] or 0,  # seconds
+        }
+
+        if self.kind == QUOTA_USER_KIND:
+            params['user'] = owner
+        elif self.kind == QUOTA_VO_KIND:
+            params['vo'] = owner
+
```
Comment (stdweird) [inline lib/vsc/filesystem/quota/tools.py:152]: else: raise Exception?

File: lib/vsc/filesystem/quota/tools.py [hunk 4]
```diff
@@ -138,22 +179,45 @@ def _push(self, storage_name, payload):
...
+                continue
+
+            user_name = user_map.get(int(user_id), None)
+            if not user_name:
+                continue
+
+            fileset_name = path_template['user'](user_name)[1]
+
+            for (fileset, quota_) in quota.quota_map.items():
+                if not re.match('^vsc[1-4]', fileset) and \
```
Comment (stdweird) [inline lib/vsc/filesystem/quota/tools.py]: why not one big regex? they all do `startswith`

File: lib/vsc/filesystem/quota/tools.py [hunk 5]
```diff
@@ -269,7 +333,7 @@ def _update_quota_entity(filesets, entity, filesystem, gpfs_quotas, timestamp, r
     @type replication_factor: int, describing the number of copies the FS holds for each file
     """
     for quota in gpfs_quotas:
-        logging.debug("gpfs_quota = %s" % (str(quota)))
+        logging.debug("gpfs_quota = %s", str(quota))
```
Comment (stdweird) [inline lib/vsc/filesystem/quota/tools.py]: no `str` needed

File: lib/vsc/filesystem/quota/tools.py (hunk 3 above)
Comment (wpoely86) [reply lib/vsc/filesystem/quota/tools.py:152]: That's already done on the actual push to the accountpage.

File: bin/dquota.py (hunk 1 above)
Comment (wpoely86) [reply bin/dquota.py]: I can change it but any other script I look at uses the same line?

Comment (stdweird) [reply bin/dquota.py]: yes, and they are all wrong. pyton in all it's brillaince allows this. it exits with 1 and prints the tuple, so it's not the exitcode we want/expect.

**Comment by wpoely86** [comment]
Remarks fixed

File: bin/dquota.py (hunk 1 above)
Comment (wdpypere) [reply bin/dquota.py]: it is not even needed. the opts.critical on the line above it will exit for you.

**Comment by wdpypere** [comment]
@wpoely86 I think you should rebase on https://github.com/hpcugent/vsc-filesystems-quota/pull/3 (or get it merged) It solves the NAGIOS_CRITICAL stuff for you.

**Comment by wpoely86** [comment]
@wdpypere Thanks. I don't have merge right on this repo so I will rebase once #3 is merged.

File: lib/vsc/filesystem/quota/tools.py [hunk 6]
```diff
@@ -38,7 +39,8 @@
 
 from collections import namedtuple
 
-from vsc.config.base import GENT_VO_PREFIX, GENT_VO_SHARED_PREFIX, STORAGE_SHARED_SUFFIX, GENT
+from vsc.config.base import STORAGE_SHARED_SUFFIX, GENT, VSC
```
Comment (itkovian) [inline lib/vsc/filesystem/quota/tools.py]: Can you do the import ( ... ) tuple thingie?

**Comment by wpoely86** [comment]
@itkovian fixed but no Jenkins it seems.

**Comment by itkovian** [comment]
Will test, sec.

**Comment by itkovian** [comment]
Should be tested, according to Jenkins :)

**Comment by itkovian** [comment]
Jenkins please test this

**Comment by itkovian** [comment]
~~~~
ok

----------------------------------------------------------------------
Ran 11 tests in 7.284s

OK
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline

Could not update commit status, please check if your scan credentials belong to a member of the organization or a collaborator of the repository and repo:status scope is selected


GitHub has been notified of this commit’s build result

Finished: SUCCESS
~~~~

**Comment by wpoely86** [comment]
This moved from github.ugent.be to github.com maybe that's the issue for Jenkins?

**Comment by wdpypere** [comment]
@wpoely86 , no it is a(nother) known problem. people are looking into it.

**Pull request merged**